web/src/App.tsx

```diff
@@ -0,0 +1,81 @@
+import { useAnalytics } from "@/analytics";
+import { Nav } from "@/components/navBar/Nav";
+import { AnalyticsDashboard } from "@/pages/analytics/AnalyticsDashboard";
+import { NotificationsMap } from "@/pages/dashboards/index";
+import { Box, Center, Flex, Spinner } from "@chakra-ui/react";
+import { createContext, Suspense } from "react";
+import { Navigate, Route, Routes } from "react-router-dom";
+import { useRecoilValue } from "recoil";
+import { Footer } from "./components/Footer";
+import {
+  CityListHookReturnType,
+  useUpdateCityList,
+} from "./data/Hooks/selectCityHook";
+import { connectionConfig } from "./data/recoil";
+import { Overview } from "./pages/configure/Overview";
+
+export let UserContext: React.Context<CityListHookReturnType>;
+
+function App() {
+  const loadingFallback = (
+    <Center height="100vh">
+      <Spinner
+        size="xl"
+        speed="0.85s"
+        thickness="3px"
+        emptyColor="gray.200"
+        color="blue.500"
+      />
+    </Center>
+  );
+
+  return (
+    <Suspense fallback={loadingFallback}>
+      <RoutesContainer />
+    </Suspense>
+  );
+}
+
+const RoutesContainer = () => {
+  const config = useRecoilValue(connectionConfig);
+  const CityListHook = useUpdateCityList(config);
+  UserContext = createContext(CityListHook);
+
+  const Analytics = () => {
```

Comment:
why do we need this component? any reason to not call `useAnalytics` in the `RoutesContainer` body?

Reply:
I think this should have been part of my cleanup

I was trying to do some stuff earlier that wanted be to decouple app and routes

But Can be reverted

Reply:
if you see useAnalytics

it is defining a fragment component 

the Suspense component has issues with fragment component and fails to use fallback with fragments as fragments are consider sibling


To avoid this there are two ways

1. wrap it with box or stack or flex instead of fragment
2. create another component that will return fragment

I don't want another wrapper block so to avoid that i went with 2

Reply:
Can we do something like this?

```

  const AnalyticsWrapper = ({ children }: { children: React.ReactElement }) => {
    useAnalytics();
    return <>
      {children}
    </>;
  };

  return (
    <Suspense fallback={loadingFallback}>
      <AnalyticsWrapper>
        <Flex height="100vh" direction="column">
          <Nav />
          <Box m={4} flex="1">
            <Routes>
              <Route path="/" element={<NotificationsMap />} />
              <Route path="/overview" element={<Overview />} />
              <Route path="/admin" element={<Admin />} />
              <Route path="/analytics" element={<AnalyticsDashboard />} />
              <Route path="*" element={<Navigate to="/" replace />} />
            </Routes>
          </Box>
        </Flex>
      </AnalyticsWrapper>
    </Suspense>
  );
  ```

Reply:
I tried and googled few things
This doesn't work because I am using hooks and function like useNavigation and useLocation
The suspense cannot be placed in function having them 

Here is one of the discussion
[Stackoverflow](https://stackoverflow.com/questions/54432861/a-react-component-suspended-while-rendering-but-no-fallback-ui-was-specified)

Suspense fallback fails implementing it in this way

to make this work I need to move out the child component logic inside another function which is what i Did